python/adjoint/reverse_design.py

```diff
@@ -0,0 +1,52 @@
+import nlopt
```

Comment:
Maybe change this to a lazy import inside the function, so that NLopt is only required if you want to run this particular function.